packages/tailwindcss/src/utilities.ts

```diff
@@ -2261,122 +2261,205 @@ export function createUtilities(theme: Theme) {
-            ]
+          switch (type) {
+            case 'angle': {
+              value = negative ? `calc(${value} * -1)` : `${value}`
+
+              return [
+                decl('--tw-gradient-position', `${value},`),
+                decl('background-image', `linear-gradient(var(--tw-gradient-stops,${value}))`),
+              ]
+            }
+            default: {
+              if (negative) return
+
+              return [
+                decl('--tw-gradient-position', `${value},`),
+                decl('background-image', `linear-gradient(var(--tw-gradient-stops,${value}))`),
+              ]
+            }
           }
         }
-      } else {
-        if (!isPositiveInteger(value)) return
 
-        value = negative ? `calc(${value}deg * -1)` : `${value}deg`
+        let value = candidate.value.value
+
+        if (!negative && linearGradientDirections.has(value)) {
+          value = linearGradientDirections.get(value)!
+        } else if (isPositiveInteger(value)) {
+          value = negative ? `calc(${value}deg * -1)` : `${value}deg`
+        } else {
+          return
+        }
+
+        let interpolationMethod = resolveInterpolationModifier(candidate.modifier)
 
         return [
-          decl('--tw-gradient-position', `${value} in oklch,`),
+          decl('--tw-gradient-position', `${value} ${interpolationMethod},`),
           decl('background-image', `linear-gradient(var(--tw-gradient-stops))`),
         ]
       }
     }
-  }
 
-  utilities.functional('-bg-linear', handleBgLinear({ negative: true }))
-  utilities.functional('bg-linear', handleBgLinear({ negative: false }))
+    utilities.functional('-bg-linear', handleBgLinear({ negative: true }))
+    utilities.functional('bg-linear', handleBgLinear({ negative: false }))
 
-  function handleBgConic({ negative }: { negative: boolean }) {
-    return (candidate: Extract<Candidate, { kind: 'functional' }>) => {
-      if (candidate.modifier) return
+    suggest('bg-linear', () => [
+      {
+        values: [...linearGradientDirections.keys()],
+        modifiers: suggestedModifiers,
+      },
+      {
+        values: ['0', '30', '60', '90', '120', '150', '180', '210', '240', '270', '300', '330'],
+        supportsNegative: true,
+        modifiers: suggestedModifiers,
+      },
+    ])
 
-      if (!candidate.value) {
-        return [
-          decl('--tw-gradient-position', `in oklch,`),
-          decl('background-image', `conic-gradient(var(--tw-gradient-stops))`),
-        ]
-      }
+    function handleBgConic({ negative }: { negative: boolean }) {
+      return (candidate: Extract<Candidate, { kind: 'functional' }>) => {
+        if (candidate.value?.kind === 'arbitrary') {
+          if (candidate.modifier) return
+          let value = candidate.value.value
+          return [
+            decl('--tw-gradient-position', `${value},`),
+            decl('background-image', `conic-gradient(var(--tw-gradient-stops,${value}))`),
```

Comment:
Should there also be a comma in the fallback?

```suggestion
            decl('background-image', `conic-gradient(var(--tw-gradient-stops,${value},))`),
```

Reply:
There shouldn't be, the comma is only because `--tw-gradient-stops` variable uses the `--tw-gradient-position` variable internally, slightly simplified it looks like this:

```
--tw-gradient-stops: var(--tw-gradient-position) var(--tw-gradient-from) var(--tw-gradient-to);
```

…where "from" and "to" are the gradient colors. So the variable includes the "," so that the resolved value looks something like:

```
--tw-gradient-stops: from 45deg, red, blue;
```

But with conic gradients the "from 45deg" part is optional, so if someone doesn't set `--tw-gradient-position` we want the resolved variable to be:

```
--tw-gradient-stops: red, blue;
```

…which is why it's important that the comma is part of `--tw-gradient-position` because it needs to be absent if that variable isn't set.

It's not needed in the fallback because if we included it you'd get a resolved conic gradient with a trailing comma which would break it:

```
background-image: conic-gradient(from 45deg,);
```

The expectation is that if you use an arbitrary conic gradient and don't use the `from-*` or `to-*` utilities, you've provided a complete conic gradient value as the arbitrary value:

```
bg-conic-[from_45deg,red,blue]
```
```

Reply:
Ohh yes makes sense, I thought this was using the variable defined above (`--tw-gradient-position`) 🤦